app/controllers/concerns/course_fetch_concern.rb

```diff
@@ -0,0 +1,41 @@
+module CourseFetchConcern
+  extend ActiveSupport::Concern
+  include ApplicationHelper
+
+private
+
+  def fetch_course
+    @course = Courses::Fetch.by_code(
+      provider_code: params[:provider_code],
+      course_code: params[:code],
+      cycle_year: cycle_year,
+    )
+  end
+
+  def fetch_courses
+    @provider = Provider
+      .includes(courses: [:accrediting_provider])
+      .where(recruitment_cycle_year: cycle_year)
+      .find(params[:provider_code])
+      .first
+
+    @courses_by_accrediting_provider = Courses::Fetch.by_accrediting_provider(@provider)
+
+    @self_accredited_courses = @courses_by_accrediting_provider.delete(@provider.provider_name)
+  end
+
+  def fetch_copy_course
```

Comment:
I think we should rename this to something like `fetch_course_to_copy_from` which will align with the `params[:copy_from]`